staging/src/k8s.io/apiserver/pkg/server/filters/reqmgmt-config.go

```diff
@@ -0,0 +1,250 @@
+
+	rmtypesv1a1 "k8s.io/api/flowcontrol/v1alpha1"
+)
+
+// initialSync does the initial setup of the configuration of the
+// reqmgmt filter.  It either succeeds and returns `true` or logs an
+// error and returns `false.
+func (reqMgmt *requestManagement) initialSync() bool {
+	rms := &RMState{
+		flowSchemas:         make(FlowSchemaSeq, 0),
+		priorityLevelStates: make(map[string]*PriorityLevelState),
+	}
+	reqMgmt.curState.Store(rms)
+	reqMgmt.plInformer.AddEventHandler(reqMgmt)
+	reqMgmt.fsInformer.AddEventHandler(reqMgmt)
+	stopCh := make(chan struct{})
+	go reqMgmt.plInformer.Run(stopCh)
+	go reqMgmt.fsInformer.Run(stopCh)
+	if ok := cache.WaitForCacheSync(stopCh, reqMgmt.plInformer.HasSynced, reqMgmt.fsInformer.HasSynced); !ok {
+		return false
+	}
+	ok := reqMgmt.syncOne()
+	if !ok {
+		klog.Error("Unable to do initial config sync")
+		return false
+	}
+	go reqMgmt.Run(stopCh)
+	return true
+}
+
+// OnAdd handles notification from an informer of object creation
+func (reqMgmt *requestManagement) OnAdd(obj interface{}) {
+	reqMgmt.configQueue.Add(0)
+}
+
+// OnUpdate handles notification from an informer of object update
+func (reqMgmt *requestManagement) OnUpdate(oldObj, newObj interface{}) {
+	reqMgmt.OnAdd(newObj)
+}
+
+// OnDelete handles notification from an informer of object deletion
+func (reqMgmt *requestManagement) OnDelete(obj interface{}) {
+	reqMgmt.OnAdd(obj)
+}
+
+func (reqMgmt *requestManagement) Run(stopCh <-chan struct{}) {
+	defer reqMgmt.configQueue.ShutDown()
+	klog.Info("Starting reqmgmt config controller")
+	go wait.Until(reqMgmt.runWorker, time.Second, stopCh)
+	klog.Info("Started reqmgmt config worker")
+	<-stopCh
+	klog.Info("Shutting down reqmgmt config worker")
+}
+
+func (reqMgmt *requestManagement) runWorker() {
+	for reqMgmt.processNextWorkItem() {
+	}
+}
+
+func (reqMgmt *requestManagement) processNextWorkItem() bool {
+	obj, shutdown := reqMgmt.configQueue.Get()
+	if shutdown {
+		return false
+	}
+
+	func(obj interface{}) {
+		defer reqMgmt.configQueue.Done(obj)
+		if !reqMgmt.syncOne() {
+			reqMgmt.configQueue.AddRateLimited(obj)
+		} else {
+			reqMgmt.configQueue.Forget(obj)
+		}
+	}(obj)
+
+	return true
+}
+
+// syncOne attempts to sync all the config for the reqmgmt filter.  It
+// either succeeds and returns `true` or logs an error and returns
+// `false`.
+func (reqMgmt *requestManagement) syncOne() bool {
+	all := labels.Everything()
+	newPLs, err := reqMgmt.plLister.List(all)
+	if err != nil {
+		klog.Errorf("Unable to list PriorityLevelConfiguration objects: %s", err.Error())
+		return false
+	}
+	newFSs, err := reqMgmt.fsLister.List(all)
+	if err != nil {
+		klog.Errorf("Unable to list FlowSchema objects: %s", err.Error())
+		return false
+	}
+	oldRMS := reqMgmt.curState.Load().(*RMState)
+	var shareSum float64
+	newRMS := &RMState{
+		priorityLevelStates: make(map[string]*PriorityLevelState, len(newPLs)),
+	}
+	newlyQuiescent := make([]*PriorityLevelState, 0)
+	var numExempt, numDefault int
```

Comment:
we dont need to count the number here. we can ensure there're only one `exempt` and only one `default` globally by validation

Reply:
Insisting on that in validation would prevent creating a new exempt priority level and deleting the old exempt one.  It would also prevent creating a new global default and deleting the old one.  The KEP allows these things to be done.